File: programmers/Lv2/84512. 모음 사전/모음 사전.py
def solution(word):
    charList = ["A", "E", "I", "O", "U"]
    count = 0
    
    def bt(result):
        nonlocal count
        count += 1
        
        if result == word:
            return True
        if len(result) == 5:
            return False
        
        for char in charList:
            if bt(result + char): 
                return True
    
    for char in charList:
        if bt(char):
            return count

File: programmers/Lv2/84512. 모음 사전/모음 사전.swift
import Foundation

func solution(_ word:String) -> Int {
    let charList = ["A", "E", "I", "O", "U"]
    var count = 0
    
    func bt(_ result: String) -> Bool {
        count += 1
        
        if result == word {
            return true
        }
        
        if result.count == 5 {
            return false
        }
        
        
        for char in charList {
            if bt(result + char) {
                return true
            }
        }
        
        return false
    }
    
    for char in charList {
        if bt(char) {
            return count
        }
    }
    
    return 0
}